test/__init__.py

```diff
@@ -100,18 +100,17 @@ def wrapper(*args, **kwargs):
         msg = "Can't run {name} because the network is unreachable".format(
             name=test.__name__)
         try:
-            return test(*args, **kwargs)
+            sock = socket.create_connection((TARPIT_HOST, 80), 0.0001)
+        except socket.timeout:
+            pass
         except socket.error as e:
             # This test needs an initial network connection to attempt the
             # connection to the TARPIT_HOST. This fails if you are in a place
             # without an Internet connection, so we skip the test in that case.
             if _is_unreachable_err(e):
                 raise SkipTest(msg)
             raise
-        except MaxRetryError as e:
-            if _is_unreachable_err(e.reason):
-                raise SkipTest(msg)
-            raise
```

Comment:
So, let's ask this question: why was this decorator not *already sufficient* for our cases? What did it not handle correctly?